### PR TITLE
WIP: Create optimization monitor classes

### PR DESCRIPTION
**Pull request by tbirdso**

### CODE REVIEW

**Review by dzenanz — COMMENTED** [review]
Maybe split into two commits: one cleaning up the unneeded example files, and the other one adding or modifying the useful files.

**Comment by tbirdso** [comment]
Refactored into two commits as suggested.

File: include/itkExhaustiveLog.h [hunk 1]
```diff
@@ -20,36 +20,36 @@
...
-{
-    <template ImageType>
-    class ExhaustiveLog<ImageType> : public itk::Command
-    {
-    public:
-        using Self = ExhaustiveLog;
-        using Superclass = itk::Command;
-        using Pointer = itk::SmartPointer<Self>;
-        itkNewMacro(Self);
+ /*
```
Comment (dzenanz) [inline include/itkExhaustiveLog.h]: 1-space indent does not seem right.

Also, why is all the code commented out?

Comment (tbirdso) [reply include/itkExhaustiveLog.h]: Code is commented in an attempt to isolate the root cause of the linking error in [Issue 2](https://github.com/InsightSoftwareConsortium/ITKOptimizationMonitor/issues/2).

It looks like VS is set up to format comments with 1-space indents. As I uncomment code I will take care to reformat accordingly.

**Comment by tbirdso** [comment]
Opening a replacement PR dedicated to singular `itkCommandExhaustiveLog` class.

**Pull request closed without merging**